installinstallmacos.py

```diff
@@ -269,6 +342,30 @@ def parse_dist(filename):
     return dist_info
 
 
+def get_board_ids(filename):
+    '''Parses a softwareupdate dist file, returning a list of supported
+    Board IDs'''
+    supported_board_ids = ""
+    with open(filename) as search:
+        for line in search:
+            line = line.rstrip()  # remove '\n' at end of line
+            if 'boardIds' in line:
+                supported_board_ids = line.split(" ")[-1][:-1]
+                return supported_board_ids
```

Comment:
This seems extremely fragile. the dist file is XML -- you should be parsing it using an XML parser.
(This also sets aside the possibility that Apple could change any of this at any time)

Reply:
I'm not sure what we would gain there. Although it's in an XML tree, it's actually in embedded JavaScript. So, sure, I could parse the output down to the specific `<scripts>` key but no further than that using XML parsing.  I've added the requirement for `var boardIds =` so it couldn't catch another instance of `boardIds`. Any other suggestions to make this more robust? Or just check that `board_id` has a value and ignore if not?

Reply:
At least make the function return something if 'var boardIds =' is never found -- ideally if Apple changed the format of the script, the code would just act as before you added any of this.